### PR TITLE
Assume form element content-type on SPA post

### PR DESCRIPTION
**Pull request by lneves12**
Some left overs from https://github.com/liferay/senna.js/pull/307

Assume the original form element content-type on the ajax post.

### CODE REVIEW

**Comment by lneves12** [comment]
@diegonvs some leftovers from my last PR. It makes sense to assume the same content-type on the Ajax request as the original form element

**Comment by diegonvs** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: src/screen/RequestScreen.js [hunk 1]
```diff
@@ -217,6 +217,7 @@ class RequestScreen extends Screen {
 		if (globals.capturedFormElement) {
 			this.addSafariXHRPolyfill();
 			body = this.getFormData(globals.capturedFormElement, globals.capturedFormButtonElement);
+			headers.add('Content-type', globals.capturedFormElement.enctype);
```
Comment (diegonvs) [inline src/screen/RequestScreen.js]:
```suggestion
			headers.append('Content-type', globals.capturedFormElement.enctype);
```

**Comment by diegonvs** [comment]
This is very weird @wincent. Running this PR on my machine `gulp test` throws an error:

```log
09 08 2019 16:53:52.574:INFO [Chrome 76.0.3809 (Mac OS X 10.14.0)]: Connected on socket V9XvpyAipL0Zh_oeAAAA with id 80500133
LOG: 'Senna was not initialized from data attributes. In order to enable its usage from data attributes try setting in the base element, e.g. `<body data-senna>`.'
ERROR: 'Failed to construct 'URL': Invalid URL tel:+0101010101'
Chrome 76.0.3809 (Mac OS X 10.14.0) RequestScreen should form navigate force post method and request body wrapped in FormData FAILED
        TypeError: headers.add is not a function
            at RequestScreen.load (src/screen/RequestScreen.js:221:12)
            at Context.<anonymous> (test/screen/RequestScreen.js:249:10)
Chrome 76.0.3809 (Mac OS X 10.14.0) RequestScreen should add submit input button value into request FormData FAILED
        TypeError: headers.add is not a function
            at RequestScreen.load (src/screen/RequestScreen.js:221:12)
            at Context.<anonymous> (test/screen/RequestScreen.js:270:10)
Chrome 76.0.3809 (Mac OS X 10.14.0) RequestScreen should navigate over same protocol the page was viewed on FAILED
        TypeError: headers.add is not a function
            at RequestScreen.load (src/screen/RequestScreen.js:221:12)
            at Context.<anonymous> (test/screen/RequestScreen.js:345:10)
ERROR: 'Failed to construct 'URL': Invalid URL tel:+999999999'
Chrome 76.0.3809 (Mac OS X 10.14.0): Executed 253 of 253 (3 FAILED) (4.032 secs / 2.249 secs)
[16:53:57] 'test:unit' errored after 8.59 s
[16:53:57] Error in plugin 'gulp-metal'
```

**Comment by wincent** [comment]
So `headers.add` is not a function... strange, seeing as `headers` [is supposedly a `MultiMap`](https://github.com/liferay/senna.js/blob/f0928640d48c0f4fe57d9518746f94aff423490a/src/screen/RequestScreen.js#L215), which does have [an `add()` implementation](https://github.com/metal/metal-multimap/blob/f1149882dbfda3c1d082a61a2cc756ac1c85a2af/src/MultiMap.js#L28).

I know that in the master branch we ripped out `MultiMap` in favor of [an actual `Headers` instance](https://github.com/liferay/senna.js/blob/0eddf7adc23b25b920a396a3965ac6a9fa1d6e11/src/screen/RequestScreen.js#L212), which [defines `set`, not `add`](https://developer.mozilla.org/en-US/docs/Web/API/Headers/Headers). I suspect that what is happening is that this PR is stale, and if you rebase this on top of master, I'd expect the `add` to fail.

**Comment by lneves12** [comment]
oh yeah, I implemented and tested this before the migration from XMLHttpRequest to Fetch, it might be not compatible and working with the current state of the master branch

**Comment by wincent** [comment]
We're about to archive this repo and mark it as read-only. As [explained here](https://github.com/liferay/liferay-frontend-projects/blob/eb83cc6c28fc68e6e70aed8a22f1b2fdf31ac277/maintenance/projects/senna/README.md) the Senna code base has been ingested into Liferay DXP itself in the form of the `frontend-js-spa-web` module, and we have migrated the v2.x history over into [the liferay-frontend-projects monorepo](https://github.com/liferay/liferay-frontend-projects) so that we can keep our stuff together (and have uniform testing, linting, building etc) and will cut any necessary maintenance releases from there.

So, if this is still relevant, would you mind sending the PR over there instead?

**Pull request closed without merging**